src/main/java/gregtech/common/metatileentities/steam/boiler/SteamLavaBoiler.java

```diff
@@ -20,15 +20,20 @@
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.FluidTank;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.List;
 
 public class SteamLavaBoiler extends SteamBoiler implements IFuelable {
 
-    private FluidTank lavaFluidTank;
+    private FluidTank fuelFluidTank;
+
+    private final List<FluidStack> boilerFuels;
```

Comment:
I would suggest using a `Map` registry. also using `<Fluid, Integer>` rather than `FluidStack`

Reply:
faster searching of recipes, avoiding duplicate recipes. and probably custom recipe may be supported in the future